modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -248,3 +249,46 @@ def from_partitions(
         frame.synchronize_labels(axis=labels_axis_to_sync)
 
     return DataFrame(query_compiler=PandasQueryCompiler(frame))
+
+
+def repartition(
+    df: Union[DataFrame, Series], axis: Optional[int] = None
+) -> Union[DataFrame, Series]:
+    """
+    Repartitioning Modin objects to get ideal partitions inside.
+
+    Allows to improve performance where the query compiler cannot yet.
+
+    Parameters
+    ----------
+    df : DataFrame or Series
+    axis : int, optional
+
+    Returns
+    -------
+    DataFrame or Series
+        The repartitioned dataframe or series, depending on the original type.
```

Comment:
can we add a section with some examples (what is bad partitioning, when one should use this function, what speed-up can re-partitioning bring) probably with a reference to [according page in modin docs](https://modin.readthedocs.io/en/stable/usage_guide/optimization_notes/index.html#understanding-modin-s-partitioning-mechanism)

Reply:
In order to be in time for the next new release, I would prefer to do it separately.

Users sometimes ask why the performance is bad, first we can ask the user how many internal partitions and offer to use this feature if necessary. By improving the performance guide, they will be able to do it without us.